x/dex/keeper/core.go

```diff
@@ -307,6 +307,12 @@ func (k Keeper) SwapCore(goCtx context.Context,
 	liqIter := NewLiquidityIterator(k, goCtx, pair, feeTiers)
 	for liqIter.HasNext() && remainingIn.GT(sdk.ZeroInt()) {
 		liq := liqIter.Next()
+
+		// break as soon as we iterated past tickLimit
+		if liq.Price().LT(msg.TickLimit) {
```

Comment:
Directionality with this inequality won't work with the generalized SwapCore, where `msg.TickLimit` may define a limit above or below the area where the trade is occurring. I would recommend writing a function `IsWithinLimitPrice(price, limitPrice, is0To1)` to resolve, but open to other approaches.